src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const checkSep = (separator === null) ? 'null' : separator;
+
+    for (let i = 0; i < this.length - 1; i++) {
```

Comment:
do you think it would be easier if you use a `for..of` loop?
```suggestion
    for (const element of this) { ...
```

Reply:
i used this type of loop cause i need to run it once less than the actual length.
thats cause at the end of the code, it checks the last element of the array, and if its falsy its not gonna added to the result.